V1 Release

cslotboom opened the pull request:
This one looks good - I had to re-base the branch sorry about #45 and #46.

This one is a big one. I had to make a few changes here and there to get the functions running.
I also made some quality of life adjustments to avoid confusing errors.

It might be worth confirming that the test suite runs on your computer, before merging...

The following significant changes were made:

**plot_model functions:**
- Error handling has been improved with a try/except block. This can avoid confusing crashes if no ODB or no model is found.
- Function now uses standardized viewport setting _setStandardViewport 
 
**Plot_deformedshape:**
- We read the closest time step instead of the exact time step. This avoids confusing crashes that can occur if the user specifies a time step not in the analysis
- We return a warning if the input time step is greater than the final time step.
- Scale is now 1 by default
- Function now uses standardized viewport setting _setStandardViewport 
 
**plot_modeshape:**
- Fixed some types
- Function now uses standardized viewport setting _setStandardViewport 
 
**_initializeFig:**
- Made a quality of life change that Allows the initial viewport to be sized based on displacement
 
**animate_fiberResponse2D/plot_fiberResponse2D:**
- Big quality of life changes.
- Functions read from output database.
 
 
**Tests:**
- Added the dev_ tag to summarize manual tests meant for developers.
- (Several tests are out of date in this category)
- Added tests for functions using the ODB
- Added a pytest suite for functions using the ODB.
- Added a readme summarizing all features explicitly tested. (All main functions from "Get_Rendering are tested")

cslotboom committed:
V1 Release

plot_model functions:
	Error handling has been improved with a try/except block
	This can avoid confusing crashes if no ODB or no model is found.

	Function now uses standardized viewport setting _setStandardViewport

Plot_deformedshape:
	We read the closest time step instead of the exact time step
	We return a warning if the input time step is greater than the final time step.
	This avoids confusing errors that can occur if the user specifies a time step not in the analysis

	Scale is now 1 by default

	Function now uses standardized viewport setting _setStandardViewport

plot_modeshape:
	Fixed some types

	Function now uses standardized viewport setting _setStandardViewport

_initializeFig
	Made a quality of life change that Allows the initial viewport to be sized based on displacement

animate_fiberResponse2D/plot_fiberResponse2D
	Big quality of life changes.

	Functions read from output database.

Tests
	Added the dev_ tag to summarize manual tests meant for developers.
	(Several tests are out of date in this category)

	Added tests for functions using the ODB

	Added a pytest suite for functions using the ODB.
Added a readme summarizing all features explicitly tested. (All main functions from "Get_Rendering are tested")

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -793,31 +746,16 @@ def nodecoordsFinal(nodetag):
-def getDispAnimationSlider(dt, Model, Loadcase, Scale = 1, 
-                           fps = 24, FrameInterval = 0, skipFrame =1, timeScale = 1,
-                           show_node_tags = 'no', show_element_tags = 'no'):
+def animate_deformedshape( Model, Loadcase, dt, Scale = 1, fps = 24, 
```
Change the input using keyword arguments to keep match with all other plot functions. 
animate_deformedshape(Model="none",LoadCase="none",dt=DefaultValue, .... )

cslotboom replied:
I can make this change so you can merge.

However I will say that these are not optional arguments - using keyword arguments implies they are optional and might confuse the user.

I do agree that we should standardize all functions to take similar inputs in a similar order though - right now **plot_model** and **plot_modeshape** stand out a little.

u-anurag replied:
You are right about confusing the users on optional arguments. My idea was to use Model="model" and LoadCase="loadcase" format everywhere, similar to other commands. This is only for these two arguments. This is also being followed in the plot_deformedshape() command. What do you suggest?

cslotboom replied:
Replied in  #53.

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -793,31 +746,16 @@ def nodecoordsFinal(nodetag):
+		return fig, ax
 
 
-def getDispAnimationSlider(dt, Model, Loadcase, Scale = 1, 
```
If this is an internal function, should we name it _getDispAnimationSlider() ?

cslotboom replied:
This function being referenced no longer exists!

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -585,29 +561,17 @@ def nodecoordsEigen(nodetag):
 	
 
-def plot_deformedshape(Model="none", LoadCase="none", tstep = -1, scale = 200, overlap='no'):
+def plot_deformedshape(Model="none", LoadCase="none", tstep = -1, scale = 1, overlap='no'):
```
If default scale factor of 200 is too big, use 10. Using scale=1 by default doesn't show any deflected shape since realistic deflections are too small as compared to the size of a structure.

cslotboom replied:
I will make this change for now.

I would argue it depends on the system being analyzed, so it might be better to allow the user to control the scale for their displacement. Doing otherwise might lead to confusion.

u-anurag replied:
I think we should have some default value here. If the users feel that the default scale factor is not suitable to their use, they can change it manually.

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -99,6 +99,7 @@ def createODB(*argv, Nmodes=0, deltaT=0.0, recorders=[]):
 	
 		for m in range(1,Nmodes+1):
 			Tarray[0,m-1]=4*asin(1.0)/(eigenVal[m-1])**0.5
+		print(Tarray, eigenVal)
```
No need to print these values.

cslotboom replied:
I will remove - I had this in there for some debugging, oops.

cslotboom committed:
Update Get_Rendering.py

The small requested changes to get rendering.
Small fixes to the docstring
Changed x to z in fibre sections.
